File: university/src/main/java/com/gutorov/university/api/news/NewsRq.java
package com.gutorov.university.api.news;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDateTime;
import java.util.UUID;

public class NewsRq {
    @NotBlank
    private String title;
    @NotBlank
    private String content;
    @NotNull
    private UUID categoryId;
    @NotNull
    private UUID authorId;
    @NotNull
    private LocalDateTime postDate;

    public NewsRq() {
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public UUID getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(UUID categoryId) {
        this.categoryId = categoryId;
    }

    public UUID getAuthorId() {
        return authorId;
    }

    public void setAuthorId(UUID authorId) {
        this.authorId = authorId;
    }

    public LocalDateTime getPostDate() {
        return postDate;
    }

    public void setPostDate(LocalDateTime postDate) {
        this.postDate = postDate;
    }
}

File: src/main/java/com/gutorov/university/api/news/NewsRq.kt
package com.gutorov.university.api.news

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime
import java.util.*

class NewsRq(
    var title: @NotBlank String? = null,
    var content: @NotBlank String? = null,
    var categoryId: @NotNull UUID? = null,
    var authorId: @NotNull UUID? = null,
    var postDate: @NotNull LocalDateTime? = null
)
